src/quacc/recipes/espresso/_base.py

```diff
@@ -218,4 +240,44 @@ def _prepare_atoms(
         **calc_flags,
     )
 
+    atoms.calc = calc
+
     return atoms
+
+
+def _prepare_copy(
+    copy_files: (
+        SourceDirectory
+        | list[SourceDirectory]
+        | dict[SourceDirectory, Filenames]
+        | None
+    ) = None,
+    calc_params: dict[str, Any] | None = None,
+    binary: str = "pw",
+) -> dict[SourceDirectory, Filenames]:
```

Comment:
If `None` is provided for `copy_files`, it seems like this function can return `None` but that is not reflected in the type hint or docstring.